### PR TITLE
Add OFTA simulation

### PR DESCRIPTION
**Pull request by NunchakusLei**
Changes,
- Added pusher-tipper (OFTA) simulation core
- Added pusher-tipper (OFTA) to sim CLI and API interfaces
- Fix some text mistakes in Wok, Runner sim related files
- Update two Wok state names

To test,
1. Activate the dev virtual environment
2. Go to <repo>/simulation directory.
3. Using CLI to interact with simulations, run
  ```bash
  python sim_cli --component {wok,runner,OFTA}
  ```
4. Using API to interact with simulations, run the flowing command. Then, go to `http://127.0.0.1:8000/docs` to play with the simulations,
  ```bash
  python sim_api [--wok-num WOK_NUM] [--sauce-bag-num SAUCE_BAG_NUM]
  ```

### CODE REVIEW

File: simulation/OFTA_simulation.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+The pusher_tipper_sim is the core of hardware-level, single Pusher-tipper group simulation. It simulates the behavior
+ of a Pusher-tipper group which will only receive I2C requests from the main controller (Raspberry Pi) and respond
+  based on which requests have been sent from the main controller. The overall workflow cycle of the Pusher-tipper
+   component is explained in the next 6 points:
+
+
+1. The Pusher-tipper will initialize in the `STANDBY` state, which will
+    - Wait for the main controller to set it activated. The main controller will only activate a Pusher-tipper group
+     if there is a cup needs to be sent to the corresponding wok associated with the Pusher-tipper group.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "if there is a cup in need to"

File: simulation/OFTA_simulation.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+2. After being set to activated, the Pusher-tipper group goes into the `TRANSPORTING` state which will
+    - Turn on the conveyor in the OFTA if cup is not been detect by the Pusher-tipper's cup sensor.
+    - Wait for the cup being transferred to the position in front of the Pusher.
+    - True off the conveyor in the OFTA once the cup is arrived the Pusher-tipper location.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "After it gets activated, the..."
"- Wait for the cup to be transferred in front of the Pusher"
"- Turn off the conveyor.... the cup has arrived to the Pusher-tipper location".

File: simulation/OFTA_simulation.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+3. Pusher-tipper will enter the `PUSHING` state once the cup is arrive. In this state it will
+    - Check if tipper is at home position.
+    - Start pushing the cup to tipper if tipper is at home, otherwise control tipper to go back home position.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: 3. .... "once the cup has arrived." 
" - Check for the position of the tipper and make sure it is on the _home_ position."
"-  Push the cup to the tipper, when tipper is in _home_ position."

File: simulation/OFTA_simulation.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+4. Pusher-tipper will enter the `TIPPING` state once the pushing is done. In thi state it will
+    - Check if the Wok is at waiting for ingredients state (that means Wok is ready).
+    - Dump the food ingredient into to Wok if it's ready.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "- Check if the Wok is at the _waiting for ingredients_ state"
"- Dispense the food ingredient into the designated wok once it is ready".

File: simulation/OFTA_simulation.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+5. After dumping the food ingredients into the Wok, Pusher-tipper group will enter `EJECTING` state which will
+    - Eject the cup from tipper into an area to wash cups.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: 5. "After dispensing the cup's contents into the wok... "
"- Eject the cup from the tipper into an area to wash the cups".

File: simulation/OFTA_simulation.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+6. Once the cup been ejected, the Pusher-tipper will cycle back to `STANDBY` state.
+
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: 6. "Once the cup has been  ejected"

File: simulation/OFTA_simulation.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+| 2            | Get state code      |       | 1        | STANDBY state
+|              |                     |       | 2        | TRANSPORTING state
+|              |                     |       | 3        | PUSHING state
+|              |                     |       | 4        | TIPPING runner state
+|              |                     |       | 5        | EJECTING state
+| 3            | Get error code      |       | 0        | No errors
+| 4            | Get Pusher-tipper request code |       | 0        | No request
+|              |                     |       | 1        | Request to activate
+|              |                     |       | 2        | Request to set Wok is ready
+|              |                     |       | 3        | Rrequest to notify if ejecting is done
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "Request to set Wok in _waiting for ingredients_ state"
"Request to notify when cup ejection is done"

File: simulation/OFTA_simulation.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+# Human Friendly Simulation
+
+You can using the following command to launch a human friendly simulation with a RESTful API interface at `http://127
+.0.0.1:8000/docs`,
+```bash
+python sim_api.py --wok-num <number_of_simulated_wok>
+```
+
+You can also using the following command to bring up the API interface without specify the wok quantity. In this case
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "You can use the following command..."

File: simulation/OFTA_simulation.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+| 3            | Request to confirm if ejecting is done  |       | 0        | Denied
+|              |                                         |       | 1        | Confirmed
+
+ Note: You can type in `stop` in the I2C simulation terminal to exit the simulation process.
+
+
+
+# Human Friendly Simulation
+
+You can using the following command to launch a human friendly simulation with a RESTful API interface at `http://127
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: "You can use the following..."

File: simulation/OFTA_simulation.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+The UI should look like the screenshot below,
+![](../docs/img/pusher_tipper_i2c_sim_001.png)
+
+
+
+
+# The End
+That's how you can use OFTA simulation.
```
Comment (johsecaba) [inline simulation/OFTA_simulation.md]: ".. the OFTA simulation."

File: simulation/OFTA_simulation.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+|              |                     |       | 3        | PUSHING state
+|              |                     |       | 4        | TIPPING runner state
+|              |                     |       | 5        | EJECTING state
+| 3            | Get error code      |       | 0        | No errors
+| 4            | Get Pusher-tipper request code |       | 0        | No request
+|              |                     |       | 1        | Request to activate
+|              |                     |       | 2        | Request to set Wok is in ready position or waiting for ingredients state
+|              |                     |       | 3        | Request to notify when cup ejection is done
+| 5            | Respond Runner request | uint8 | 0        | Failed to save data and/or not able to setup
+|              |                     |       | 1        | Successfully save data and/or setup
```
Comment (migueltrevinom) [inline simulation/OFTA_simulation.md]: what's the request 6 exactly doing? I don't understand, I tried to test it out and see which impact does by putting 1 or 0 but I couldn't figure it out

Comment (migueltrevinom) [reply simulation/OFTA_simulation.md]: clarified now.

**Pull request merged**